### ISSUE TITLE
databaseZone, javaZone not honored in 3.1.0CR9/3.1.0CR10

### ISSUE THREAD

**Issue by ironcream** [JadiraOrg/jadira#11]
This issue could be considered a reopen of a [JDF-71](https://jadira.atlassian.net/browse/JDF-71) since I was unable to reopen it or report anything in JIRA.

Versions prior to the commit below do not have this issue:
[https://github.com/JadiraOrg/jadira/commit/a9bd21f]

Following libs and software are in use in project, where issue is reproducible:
joda-time 2.3
spring-data-jpa 1.4.2.RELEASE
hibernate-entitymanager, hibernate-core 4.2.1.Final
mysql-connector-java 5.1.29 (most recent available, there is no such version as 6.1.6 in repos anywhere)

JVM ( "1.7.0_45" Java HotSpot(TM) 64-Bit Server VM (build 24.45-b08, mixed mode)) is running in UTC+2
MySQL DB (5.5.35) is running in UTC+2

DDL of a table where data is to be stored looks like this:

``` SQL
ALTER TABLE `something` ADD COLUMN (`date_created_utc` DATETIME DEFAULT NULL, `date_updated_utc` DATETIME DEFAULT NULL);
```

Mapping for entities looks like this:

``` java
    @Column(name = "date_created_utc")
    @Type(type = "org.jadira.usertype.dateandtime.joda.PersistentDateTime",
            parameters = {
                    @Parameter(name = "databaseZone", value = "UTC"),
                    @Parameter(name = "javaZone", value = "UTC")
            })
    protected DateTime dateCreatedUtc = DateTime.now(DateTimeZone.forID("UTC"));

    @Column(name = "date_updated_utc")
    @Type(type = "org.jadira.usertype.dateandtime.joda.PersistentDateTime",
            parameters = {
                    @Parameter(name = "databaseZone", value = "UTC"),
                    @Parameter(name = "javaZone", value = "UTC")
            })
    protected DateTime dateUpdatedUtc;

    @PreUpdate
    public void preUpdate() {
        dateUpdatedUtc = DateTime.now(DateTimeZone.forID("UTC"));
    }
```
# Expected/Desired behaviour

is that DB wil have timestamps in UTC stored when such entities get created/updated.
I.e. I execute create and update at 13:10 (UTC+2) and the value I expect to see in the database is 11:10(which I know is correct UTC equivalent of that moment of time).
# Actual behaviour

However value 13:10 is inserted. Which is supported by logs and could be checked via performing select from console.

In other words attempts to save or update such entities result in wrong timestamps written into the database (according to the DB log of incoming requests).

The strange thing is that reading such a timestamp back from the entity via spring-data-jpa repositories still produces correct DateTime, which is 11:10 with the TZ set to UTC.

Using version of jadira prior to [https://github.com/JadiraOrg/jadira/commit/a9bd21f] results in expected behaviour and does not produce issue described.
